File: credit.c
#include <stdio.h>
#include <cs50.h>

int main(void)
{
    unsigned long long numer = get_long("Number: ");
    //printf("%llu\n",numer);
    unsigned  long ten=10000000000; //ten zero
    ////////////////////////////
    unsigned long long num=numer;
    int a=0;
    int suma=0;
    int b=0;
    int c=0;
    int sumb=0;
    int sum=0;
   //////////////////////////////////////////
   for(int i=1;i<=8;i++){
   a=num % 10;
   suma=suma+a;
   num=num-a;
   num=num/10;
   
   c=num % 10;
   b=c*2;
   sumb=sumb+b;
   num=num-c;
   num=num/10;
   }
   
   sum=suma+sumb;
   if(sum%10!=0){
       printf("INVALID\n");
       return 0;
   }
   //////////////////////////////////////////////
    
    if (numer<999999999999){
        printf("INVALID\n");
    }
    //else if ((numer>=(100*ten)&&numer<(1000*ten))||(numer>=(10000*ten)&&numer<(100000*ten))){//Visa uses 13- and 16-digit numbers
    else if ((numer>=(400*ten)&&numer<(500*ten))||(numer>=(400000*ten)&&numer<(500000*ten))){//and all Visa numbers start with 4
        printf("VISA\n");
    }                                                                                                   //MasterCard uses 16-digit numbers
    else if(numer>=(510000*ten)&&numer<(560000*ten)){      //most MasterCard numbers start with 51, 52, 53, 54, or 55
            printf("MASTERCARD\n");                             
    }
    else if (numer>=(10000*ten)&&numer<(10000*ten)){//American Express uses 15-digit numbers,
        printf("American Express???\n");
    }
    else if ((numer>=(340000000000000)&&numer<(350000000000000))||(numer>=(370000000000000)&&numer<(380000000000000))){//All American Express numbers start with 34 or 37;
        printf("AMEX\n");
    }
    else{
        printf("INVALID\n");
    }
}

File: credit.py
from cs50 import get_int
number=get_int("Number: ")
##number=int(input("Number: "))

num=number
sum_multiply=0
sum_digits=0
lastNumber=0

for i in range(8):

    lastNumber=num%10;
    sum_digits=+lastNumber
    num=-lastNumber
    num=num/10
    
    lastNumber=num%10
    sum_multiply=+lastNumber*2
    num=-lastNumber
    num=num/10
    
sum=sum_digits+sum_multiply

if number<999999999999: print("INVALID\n")
elif (number>=4*10**12 and number<5*10**12)or(number>=4*10**15 and number<5*10**15):
    if sum%10!=0: print("INVALID\n")
    else: print("VISA\n")
elif number>=51*10**13 and number<56*10**13: print("MASTERCARD\n")
elif (number>=34*10**13 and number<35*10**13)or(number>=37*10**13 and number<38*10**13): print("AMEX\n")
elif number>=10**14 and number<10**15: print("American Express\n")
else: print("INVALID\n")







